lib/msal-node/src/client/ClientApplication.ts

```diff
@@ -346,16 +347,21 @@ export abstract class ClientApplication {
 
         serverTelemetryManager?.updateRegionDiscoveryMetadata(discoveredAuthority.regionDiscoveryMetadata);
 
+        // if cache is persisted, restore cache context if needed before building client configuration
+        if (this.config.cache.cachePlugin) {
```

Comment:
Does setting a `cachePlugin` mean it is `shared`? I understand that is how we have it today, but do we have a case where we can different `persistent` and `shared` cache where we need not read/write to the disk ALWAYS if the cache isn't shared.

Can be a separate PR and also point to note for web apps and cache, cc @bgavrilMS

Reply:
Same thing will help with not sticking this code in `getAllAccounts()` every time it is called and the user chose a Disk cache option.

Reply:
I suppose this could be an option, in particular for performance. It would be equivalent to saying "populate the in-memo cache with disk cache when its empty, but then let in-memo cache be the single source of truth after". In that case, if the disk cache is modified outside of msal, msal's in-memo cache will diverge from the disk cache until tokens in the in-memo cache expires (after that, msal acquires tokens from network, then syncs in-memo cache again with disk cache in `handleServerTokenResponse`)